File: admin/src/api/system_log_sms.ts
import request from '@/utils/request'
import type { Pages } from '@/utils/request'

import config from '@/config'
import queryString from 'query-string'
import { getToken } from '@/utils/auth'
import { clearEmpty } from '@/utils/util'

export type type_system_log_sms = {
    Id?: number
    Scene?: number
    Mobile?: string
    Content?: string
    Status?: number
    Results?: string
    SendTime?: string
    CreateTime?: string
    UpdateTime?: string
}
// 查询
export type type_system_log_sms_query = {
    Scene?: number
    Mobile?: string
    Content?: string
    Status?: number
    Results?: string
    SendTime_start?: string
    SendTime_end?: string
    CreateTime_start?: string
    CreateTime_end?: string
    UpdateTime_start?: string
    UpdateTime_end?: string
}
// 添加编辑
export type type_system_log_sms_edit = {
    Id?: number
    Scene?: number
    Mobile?: string
    Content?: string
    Status?: number
    Results?: string
    SendTime?: string
}

// 系统短信日志列表
export function system_log_sms_list(params?: type_system_log_sms_query) {
    return request.get<Pages<type_system_log_sms>>({
        url: '/system_log_sms/list',
        params: clearEmpty(params)
    })
}
// 系统短信日志列表-所有
export function system_log_sms_list_all(params?: type_system_log_sms_query) {
    return request.get<type_system_log_sms[]>({
        url: '/system_log_sms/listAll',
        params: clearEmpty(params)
    })
}

// 系统短信日志详情
export function system_log_sms_detail(id: number | string) {
    return request.get<type_system_log_sms>({ url: '/system_log_sms/detail', params: { id } })
}

// 系统短信日志新增
export function system_log_sms_add(data: type_system_log_sms_edit) {
    return request.post<null>({ url: '/system_log_sms/add', data })
}

// 系统短信日志编辑
export function system_log_sms_edit(data: type_system_log_sms_edit) {
    return request.post<null>({
        url: '/system_log_sms/edit',
        data: data
        // data: clearEmpty(data)
    })
}

// 系统短信日志删除
export function system_log_sms_delete(id: number | string) {
    return request.post<null>({ url: '/system_log_sms/del', data: { id } })
}

// 系统短信日志导入
export const system_log_sms_import_file = '/system_log_sms/ImportFile'

// 系统短信日志导出
export function system_log_sms_export_file(params: any) {
    return (window.location.href =
        `${config.baseUrl}${config.urlPrefix}/system_log_sms/ExportFile?token=${getToken()}&` +
        queryString.stringify(clearEmpty(params)))
}

File: server/model/system_log_sms.go
package model

import (
	"x_admin/core"
)

// SystemLogSms 系统短信日志实体
type SystemLogSms struct {
	Id         int           `gorm:"primarykey;comment:'id'"`                // id
	Scene      core.NullInt  `gorm:"comment:'场景编号'"`                         // 场景编号
	Mobile     string        `gorm:"comment:'手机号码'"`                         // 手机号码
	Content    string        `gorm:"comment:'发送内容'"`                         // 发送内容
	Status     core.NullInt  `gorm:"comment:'发送状态：[0=发送中, 1=发送成功, 2=发送失败]'"` // 发送状态：[0=发送中, 1=发送成功, 2=发送失败]
	Results    string        `gorm:"comment:'短信结果'"`                         // 短信结果
	SendTime   core.NullTime `gorm:"comment:'发送时间'"`                         // 发送时间
	CreateTime core.NullTime `gorm:"autoCreateTime;comment:'创建时间'"`          // 创建时间
	UpdateTime core.NullTime `gorm:"autoUpdateTime;comment:'更新时间'"`          // 更新时间
}
